cpg-core/src/main/kotlin/de/fraunhofer/aisec/cpg/passes/ControlFlowSensitiveDFGPass.kt

```diff
@@ -175,6 +177,40 @@ open class ControlFlowSensitiveDFGPass(ctx: TranslationContext) : EOGStarterPass
                     PowersetLattice(identitySetOf(currentNode))
                 )
             }
+        } else if (currentNode is MemberExpression && currentNode.access == AccessValues.WRITE) {
+            // already set in DFG pass, because otherwise we cannot set the field property
+            // state.push(currentNode.base, PowersetLattice(identitySetOf(currentNode)))
+
+            writtenDeclaration = (currentNode.base as? Reference)?.refersTo
+
+            if (writtenDeclaration != null) {
```

Comment:
Does this work for accesses like `base.a.b`?

Reply:
The answer is "jein". It works to some degree, but not perfect. I wanted to add another unit test for this once we agree on the general direction of this PR.

Reply:
I am not quite sure if the dataflow of the nested structures are really 100 % "correct" in this case. Its not easy to imagine how they "should" look like.

Reply:
I added a "full" edge between the occurrences of `outer::i` and also added a "partial" edge from `outer::i` to `o`.